crates/kubelet/src/state/common/mod.rs

```diff
@@ -50,21 +52,21 @@ pub trait GenericProviderState: 'static + Send + Sync {
 /// Exposes pod state in a way that can be consumed by
 /// the generic states.
 #[async_trait::async_trait]
-pub trait GenericPodState: 'static + Send + Sync {
+pub trait GenericPodState: ResourceState<Manifest = Pod, Status = PodStatus> {
     /// Stores the pod module binaries for future execution. Typically your
     /// implementation can just move the modules map into a member field.
-    fn set_modules(&mut self, modules: HashMap<String, Vec<u8>>);
+    async fn set_modules(&mut self, modules: HashMap<String, Vec<u8>>);
```

Comment:
So, I ran into this because the `ModuleRunContext` had to be moved into a `SharedState` so that individual container state machines could load their module data. I'm glad you brought this up because I think that we should make pretty much all public trait methods `async`, the reason being that we have a lot of `async` methods and Tokio `RwLocks` floating around and it is really annoying as a provider author to realize that you can't use something `async` in an `async` trait just because we (the maintainers) didn't forsee that use case. Thoughts?

Reply:
I figured that may be part of the reasoning behind this and I am fine with it. But I also try to avoid having things be async if they don't have to be. The safe bet is to probably keep it async and then we can pull that back before 1.0 if it truly isn't being used as async

Reply:
Sounds good.